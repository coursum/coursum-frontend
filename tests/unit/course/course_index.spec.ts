import CourseIndex from '@/components/course/course_index.vue';

import { shallowMount, mount } from '@vue/test-utils';

jest.mock('@/api/request');

describe('CourseIndex', () => {
  it('should render correct markup when prop is correct', () => {
    const wrapper = shallowMount(CourseIndex);

    expect(wrapper.element).toMatchSnapshot();
  });

  it('should render correct markup when prop is correct', () => {
    const wrapper = mount(CourseIndex);

    expect(wrapper.element).toMatchSnapshot();
  });
});
