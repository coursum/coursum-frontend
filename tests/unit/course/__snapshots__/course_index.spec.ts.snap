// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseIndex should render correct markup when prop is correct 1`] = `
<div>
  <div
    class="d-flex flex-row-reverse pt-3 pr-3"
  >
    <setting-button-stub />
     
    <account-button-stub />
  </div>
   
  <div
    class="mx-auto text-center"
  >
    <coursum-logo-stub />
     
    <search-input-stub />
  </div>
   
  <div
    class="mx-0 my-6"
  >
    <div
      class="d-flex flex-wrap justify-space-around px-3"
    >
      <course-show-stub
        coursedata="[object Object]"
        haswidth="true"
        showsummary="true"
        texttruncate="true"
      />
      <course-show-stub
        haswidth="true"
        showsummary="true"
        texttruncate="true"
      />
      <course-show-stub
        haswidth="true"
        showsummary="true"
        texttruncate="true"
      />
      <course-show-stub
        haswidth="true"
        showsummary="true"
        texttruncate="true"
      />
      <course-show-stub
        haswidth="true"
        showsummary="true"
        texttruncate="true"
      />
      <course-show-stub
        haswidth="true"
        showsummary="true"
        texttruncate="true"
      />
      <course-show-stub
        haswidth="true"
        showsummary="true"
        texttruncate="true"
      />
      <course-show-stub
        haswidth="true"
        showsummary="true"
        texttruncate="true"
      />
      <course-show-stub
        haswidth="true"
        showsummary="true"
        texttruncate="true"
      />
      <course-show-stub
        haswidth="true"
        showsummary="true"
        texttruncate="true"
      />
      <course-show-stub
        haswidth="true"
        showsummary="true"
        texttruncate="true"
      />
      <course-show-stub
        haswidth="true"
        showsummary="true"
        texttruncate="true"
      />
    </div>
     
    <div
      class="mx-auto my-6"
      style="max-width: 70%;"
    >
      <v-pagination-stub
        class="mb-12"
        currentpagearialabel="$vuetify.pagination.ariaLabel.currentPage"
        length="1"
        nextarialabel="$vuetify.pagination.ariaLabel.next"
        nexticon="$next"
        pagearialabel="$vuetify.pagination.ariaLabel.page"
        previcon="$prev"
        previousarialabel="$vuetify.pagination.ariaLabel.previous"
        value="1"
        wrapperarialabel="$vuetify.pagination.ariaLabel.wrapper"
      />
    </div>
  </div>
</div>
`;
